contracts/programs/bubblegum/src/lib.rs

```diff
@@ -37,30 +38,88 @@ pub struct CreateTree<'info> {
     init,
     seeds = [merkle_slab.key().as_ref()],
     payer = payer,
-    space = NONCE_SIZE,
+    space = GUMMYROLL_TREE_AUTHORITY_SIZE,
     bump,
     )]
-    pub authority: Account<'info, Nonce>,
+    pub authority: Account<'info, GummyrollTreeAuthority>,
     #[account(mut)]
     pub payer: Signer<'info>,
     pub tree_creator: Signer<'info>,
     pub candy_wrapper: Program<'info, CandyWrapper>,
-    pub system_program: Program<'info, System>,
     pub gummyroll_program: Program<'info, Gummyroll>,
     #[account(zero)]
     /// CHECK: This account must be all zeros
     pub merkle_slab: UncheckedAccount<'info>,
+    pub system_program: Program<'info, System>,
+}
+
+#[derive(Accounts)]
+pub struct TransferTreeOwner<'info> {
+    pub tree_owner: Signer<'info>,
+    /// CHECK: this account is not read from
+    pub new_owner: AccountInfo<'info>,
+    #[account(
+        seeds=[authority.tree_id.as_ref()],
+        bump,
+        mut, 
+        constraint = *tree_owner.key == authority.owner
+    )]
+    pub authority: Account<'info, GummyrollTreeAuthority>,
+}
+
+#[derive(Accounts)]
+pub struct TransferTreeDelegate<'info> {
+    pub tree_delegate: Signer<'info>,
+    /// CHECK: this account is not read from
+    pub new_delegate: AccountInfo<'info>,
+    #[account(
+        mut,
+        seeds=[authority.tree_id.as_ref()],
+        bump,
+        constraint = (*tree_delegate.key == authority.owner) || (*tree_delegate.key == authority.delegate)
+    )]
+    pub authority: Account<'info, GummyrollTreeAuthority>,
+}
+
+#[derive(Accounts)]
+pub struct SetTreeAppendAuthority<'info> {
```

Comment:
I think this workflow is pretty tricky. I wonder if it's the right approach to something like:
1) User requests to be an append_authority with a certain size
2) Tree creator needs to approve the request and append_authority gets granted N mints
3) Tree creator can be a governance thing in the future because decentralization or something like that

Reply:
Tree creator should be like super admin. Should be able to do anything (abstracts away authorization to higher contract like gov if necessary)

Reply:
If this is the case then tree creator should have not the ability to change the delegate